libs/pushstreams/api/celix/impl/AbstractPushEventSource.h

```diff
@@ -102,15 +102,15 @@ void celix::AbstractPushEventSource<T>::open(std::shared_ptr<celix::IPushEventCo
 
 template <typename T>
 [[nodiscard]] celix::Promise<void> celix::AbstractPushEventSource<T>::connectPromise() {
-    std::lock_guard lck{mutex};
+    std::unique_lock lck{mutex};
```

Comment:
why unique_lock? IMO lock_guard is preferred If no condition_variable is used (call to wait) in the function/scope. 
So in this case only on line 159 is unique_lock needed, the rest van be kept a lock_guard

Reply:
done